server/controllers/profile.js

```diff
@@ -0,0 +1,118 @@
+const Profile = require("../models/Profile");
+const User = require("../models/User");
+const asyncHandler = require("express-async-handler");
+
+exports.getProfile = asyncHandler(async (req, res, next) => {
+  const id = req.body.id;
+
+  const profile = await Profile.find(
```

Comment:
Why don't we consider using a `findById` query here instead 

Ref: https://mongoosejs.com/docs/api.html#model_Model.findById